python/eos/plot/plotter.py

```diff
@@ -283,8 +283,27 @@ def __init__(self, plotter, item):
         def plot(self):
             item = self.item
             oname = item['observable']
+            eos.Observables.assert_valid_name(oname)
             eos.info('   plotting EOS observable "{}"'.format(oname))
 
+            # handle plot variable, create kinematic or parameter
+            if not 'variable' in item:
+                raise ValueError("Missing key for plot of observable '" + oname + "': 'variable'")
+            try:
+                eos.Parameters.assert_valid_name(item['variable'])
+                var = parameters.declare(item['variable'], np.nan)
+            except ValueError:
+                eos.Observables.assert_valid_name_and_kinematic(oname, item['variable'])
+                var = parameters.declare(item['variable'], np.nan)
+            finally:
+                raise ValueError("Value of 'variable' for observable '" + oname +
+                    "' is neither a valid kinematic variable nor parameter: '" + item['variable'] + "'")
+
+            # create kinematics
+            if 'kinematics' in item:
+                for k, v in item['kinematics'].items():
```

Comment:
You should probably check if the variable was a kinematic variable, and avoid overriding it here.This should probably be an error.

Reply:
Implemented.